waku/node/peer_manager/peer_manager.nim

```diff
@@ -315,6 +321,39 @@ proc onPeerEvent(pm: PeerManager, peerId: PeerId, event: PeerEvent) {.async.} =
     direction = if event.initiator: Outbound else: Inbound
     connectedness = Connected
 
+    # To prevent metadata protocol from breaking prev nodes, by now we only
+    # disconnect if the clusterid is specified.
+    if not pm.wakuMetadata.isNil() and pm.wakuMetadata.clusterId != 0:
+      var conn: Connection
+      try:
+        conn = await pm.switch.dial(peerId, WakuMetadataCodec)
+      except CatchableError:
+        info "disconnecting from peer", peerId=peerId, reason="waku metadata codec not supported"
+        asyncSpawn(pm.switch.disconnect(peerId))
```

Comment:
Without having looked into the details it's not immediately clear to me if this `disconnect` would also trigger a proper cleanup of the peer in the peer store/manager so that we don't keep stale peers?

Reply:
indeed, good catch. removing the peer here:
dbd692ac606c80392a47f4dc0c27523b167f7859